src/action.ts

```diff
@@ -11,8 +11,14 @@ async function main() {
       }
     })
     let settings = lib.parseSettings(env)
-    await lib.main(settings)
-  } catch(e) {
+    const pr = await lib.main(settings)
+    if (pr != null) {
+      console.log(`Setting output pr_number=${pr.number}`)
+      core.setOutput('pr_number', pr.number.toString())
+      console.log(`Setting output pr_url=${pr.url}`)
+      core.setOutput('pr_url', pr.url)
```

Comment:
Nit: we can drop the logging lines, we already log the PR URL when we create/update it

Reply:
I guess it depends how much you trust the program.

Since Github has no validation on any output, you can specify `steps.upgrade.outputs.pr` and it won't error, it can be hard to trace issues especially since the setting of this output is conditional.

I don't mind removing them, but i have found it helpful when working with other actions to know what they are doing

Reply:
Ah yep, so it's more about debugging the keys that are being set rather than the values. Makes sense 👍